src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/NoWriteNoSeekStreamContent.cs

```diff
@@ -26,15 +26,18 @@ internal NoWriteNoSeekStreamContent(Stream content)
             _content = content;
         }
 
-        protected override Task SerializeToStreamAsync(Stream stream, TransportContext? context) =>
-            SerializeToStreamAsync(stream, context, CancellationToken.None);
-
 #if NETCOREAPP
-        protected override
-#else
-        internal
+        protected override void SerializeToStream(Stream stream, TransportContext? context, CancellationToken cancellationToken) =>
+            SerializeToStreamAsyncCore(stream, context, async: false, cancellationToken).GetAwaiter().GetResult();
+
+        protected override void SerializeToStreamAsync(Stream stream, TransportContext context, CancellationToken cancellationToken) =>
+            SerializeToStreamAsyncCore(stream, context, async: true, cancellationToken);
 #endif
-            Task SerializeToStreamAsync(Stream stream, TransportContext? context, CancellationToken cancellationToken)
+
+        protected override Task SerializeToStreamAsync(Stream stream, TransportContext? context) =>
+            SerializeToStreamAsyncCore(stream, context, async: true, CancellationToken.None);
+
+        private Task SerializeToStreamAsyncCore(Stream stream, TransportContext? context, bool async, CancellationToken cancellationToken)
```

Comment:
We probably do not need sync support in WinHttpHandler, I can remove it if we don't want it here.

Reply:
Do we even build WinHttpHandler for NetCoreAppCurrent?

Reply:
I doesn't seem so: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Http.WinHttpHandler/src/System.Net.Http.WinHttpHandler.csproj#L5
The whole  `#if NETCOREAPP` seems to be just a remnant from being part of Http dll and could be removed as well.

Reply:
Let's remove it then.

Reply:
Reverted WinHttpHandler changes.